app/signIn/page.tsx

```diff
@@ -1,23 +1,37 @@
 'use client';
 import {useForm} from "react-hook-form";
 import pb from "@/app/lib/pocketbase"
-import {useState} from "react";
+import {useContext, useEffect, useState} from "react";
 import {randomInt} from "node:crypto";
-
+import UserContext from "@/app/userContext"
 
 export default function SignIn() {
     const {register, handleSubmit, reset} = useForm();
     const [isLoading, setIsLoading] = useState<boolean>(false);
     const [isLogged, setIsLogged] = useState<boolean>(false);
-    const [dummy ,setDummy] = useState(0);
+    const [dummy, setDummy] = useState(0);
+    const [user, setUser] = useContext(UserContext);
+
+    useEffect(() => {
+        // Check if the user is already logged in from localStorage
+        const storedUser = localStorage.getItem('user');
+        if (storedUser) {
+            setUser(storedUser);
+            setIsLogged(true);
+        }
+    }, []);
+
     async function signInUser(data: any) {
         setIsLoading(true);
         try {
             const authData = await pb.collection('users').authWithPassword(
                 data.username,
                 data.password,
             );
-            setIsLogged(pb.authStore.isValid);
+            setIsLogged(true);
```

Comment:
separate hook that exposes a handler for that could be done, extending on the code above like:

```
const isLoggedIn = () => {
   const [isLoggedIn, setIsLoggedIn] = useState<boolean>(false);
   const [user, setUser] = useContext(UserContext);
   
   useEffect(() => {
        const storedUser = localStorage.getItem('user');
        if (storedUser) {
            setUser(storedUser);
            setIsLoggedIn(true);
        }
    }, []);
    
    handleLogIn = () => {
        setIsLoggedIn(true);
        setUser(pb.authStore.model?.id);
        localStorage.setItem('user', pb.authStore.model?.id);
     }
    
    handleLogOut = () => {
      setIsLoggedIn(true);
      setUser(null);
      localStorage.removeItem('user');
    }

    return {
      isLoggedIn,
      onLogIn: handleLogIn,
      onLogOut: handleLogout
   }   
}
```